tests/federationapi_sendtodevice_test.go

```diff
@@ -0,0 +1,194 @@
+package tests
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+)
+
+func TestSendToDevice(t *testing.T) {
```

Comment:
at a glance, it looks like this is pretty similar to the C-S version, so my comments apply here too.